dist/index.js

```diff
@@ -0,0 +1,144 @@
+    string: ['config_path']
+});
+
+(function () {
+    const elmDocstyleVersion = require(path.join(__dirname, '../', 'package.json')).version;
+    if (args.help) {
+        console.log(helpStr);
+        process.exit(1);
+    }
+
+    if (args._.length === 0) {
+        console.log("Please specify at least one directory or path to Elm source file.\n");
+        console.log(helpStr);
+        process.exit(1);
+    }
+
+    if (args.version) {
+        console.log(elmDocstyleVersion);
+        process.exit(0);
+    }
+
+    const validFormats = ['json', 'human'];
+
+    var excludedChecks = [];
+
+    if (args.config_path !== undefined) {
+        try {
+            const txt = fs.readFileSync(args.config_path, 'utf8');
+
+            const excluded = (JSON.parse(txt)).excludedChecks;
+            if (excluded === undefined || (excluded instanceof Array) === false
+                || (excluded.length > 0 && (typeof excluded[0] !== "string"))) {
+                console.log("the provided config path does not contain a valid elm-docstyle configuration.");
+                console.log("Please refer to the README to see how a correct config file is shaped.");
+                process.exit(1);
+            }
+            excludedChecks = excluded;
+        } catch (err) {
+            console.log("failed to open the input config_path:");
+            console.log(err);
+            process.exit(1);
+        }
+    }
+
+    const checker = Elm.Main.worker({
+        format: validFormats.indexOf(args.format) !== -1 ? args.format : 'human'
+        , verbose: args.verbose || false
+        , excludedChecks: excludedChecks
+        , checkAllDefinitions: args.checkAllDefinitions || false
+    });
+
+
+    var reportList = [];
+    checker.ports.outgoing.subscribe(function (newReport) {
+        reportList.push(newReport);
+    });
+
+    // explore recursively and find all elm files
+    var exploreAndSend = function (pth) {
+        if (pth.endsWith(".elm")) {
+            try {
+                const txt = fs.readFileSync(pth, 'utf8');
+                checker.ports.incoming.send(txt);
+            } catch (err) {
+                console.log("failed to open the input path " + pth + ":");
+                console.log(err);
+                process.exit(1);
+            }
+        } else if (pth.endsWith("/")) {
+            if (pth === "elm-stuff"){
+                return
+            }
+            var paths = [];
+            try {
+                paths = fs.readdirSync(pth, 'utf8');
+            } catch (err) {
+                console.log("failed to open the input dir " + pth + ":");
+                console.log(err);
+                process.exit(1);
+            }
+            paths.map(path => pth + path).forEach(a => exploreAndSend(a));
+        }
+    };
+
+    args._.forEach(a => exploreAndSend(a));
+
+    // in milliseconds
+    const timeoutStep = 50;
+
+    function finalCheck(roundNr) {
+        if (roundNr > 60000 / timeoutStep) {
```

Comment:
What if the directory tree is very deep? Why is it needed not to run longer than 60 seconds? (This is something that at least needs to be documented and explained in the error message; I would be totally perplexed as a user :-))

Reply:
I just put this as a security measure, as I was scared of looping indefenitely. Should I remove it altogether?